src/scripts/sniperino.ts

```diff
@@ -0,0 +1,139 @@
+// Description:
+//  Beat the sniperino
+//  
+//
+// Commands:
+//  sniperino( me)? - start a new game of sniperino 
+//  snipe( me)? - roll a snipe roll in an attempt to beat the sniperino
+//
+// Author:
+//  Steve Shipsey
+
+import { Robot } from "tsbot";
+
+interface ISniper {
+    name: string;
+    gamesPlayed: number;
+    gamesWon: number;
+    currentSnipe?: number;
+
+    getWinRate(): number;
+    generateNewSnipe();
+}
+
+class Sniper implements ISniper {
+    public name: string;
+    public gamesPlayed: number;
+    public gamesWon: number;
+    public currentSnipe: number;
+    
+    constructor(n: string) {
+        this.name = n.slice(0, n.indexOf(" "));
+        this.gamesPlayed = 0;
+        this.gamesWon = 0;
+    }
+
+    public getWinRate(): number {
+        // Calculation with some javascript rounding magic
+        //return this.gamesPlayed > 0 ? Number(Math.round((this.gamesWon / this.gamesPlayed * 100) + 'e2') + 'e-2') : 0;
+        return this.gamesPlayed > 0 ? Math.round(this.gamesWon / this.gamesPlayed * 100 * Math.pow(10, 2)) / Math.pow(10,2) : 0;
```

Comment:
Math.pow(10, 2) doesn't need to be evaluated twice... also its a static number why do the math at all?